source/server/http/admin.cc

```diff
@@ -402,11 +404,21 @@ Http::Code AdminImpl::handlerStats(absl::string_view url, Http::HeaderMap& respo
     all_stats.emplace(gauge->name(), gauge->value());
   }
 
+  for (const Stats::HistogramSharedPtr& histogram : server_.stats().histograms()) {
+    all_histograms.emplace(histogram->name(),
```

Comment:
this pattern is pre-existing, so extending it is OK.  But IMO if the goal is to sort, populate a vector and use std::sort, rather than inserting into a map and pulling it out.   Two reasons:
 1a. faster -- both std::map and std::sort are n*log(n) but sort has a better constant factor.
 1b. Even more faster if they are kept sorted to begin :)
 2. I think it might make sense to include the same named histogram from multiple scopes, and this would make only the *last* one win, whereas other logic I've seen above makes the *first* one win.   Arguably they all should be included.  Maybe in that case incorporate the scope as well into the key, probably more significant than the name.  In the case where you are seeing the dup-warnings (previously asserts) I think this last-wins behavior might be wrong.

WDYT about at least adding a TODO to clean this up, for this and also all_stats above?

Reply:
Yes we can/should clean this up.

I've explained the reason there are de-dup issues. This is a non-issue for counters/gauges as they are stored in shared memory. This is an issue for histograms since they don't share storage. I think ultimately we need process global storage for histograms similar to how we have a central storage in shared memory for counters/gauges. I think we can do this in a follow up, but I would clearly lay out in a comment/TODO what the issue is.